screeps-game-api/src/macros.rs

```diff
@@ -482,6 +482,36 @@ macro_rules! impl_serialize_as_i32 {
     };
 }
 
+/// Implements `Iterator` for `js_vec::IntoIter` or `js_vec::Iter`.
+macro_rules! impl_js_vec_iterators {
+    ($($name:ident $(<$single_life_param:lifetime>)*),* $(,)*) => {
+        $(
+            impl<$($single_life_param, )* T> Iterator for $name<$($single_life_param, )* T>
+            where
+                T: TryFrom<Value, Error = ConversionError>,
+            {
+                type Item = Result<T, ConversionError>;
+
+                fn next(&mut self) -> Option<Self::Item> {
+                    if self.index as usize >= self.inner.len() {
+                        None
+                    } else {
+                        let index = self.index;
+                        self.index = self
+                            .index
+                            .checked_add(1)
+                            .expect("bounds overflow in JsVec IntoIter");
```

Comment:
Do you expect longer vector than `usize` that much?

Reply:
No, but I was trying to cover all cases. Probably silly since we're 32 bit in any case and there's the check above for the length.